recipes-web/iot-rest-api-server/iot-rest-api-server.bb

```diff
@@ -27,8 +27,7 @@ SYSTEMD_AUTO_ENABLE_${PN} = "disable"
 USERADD_PACKAGES = "${PN}"
 GROUPADD_PARAM_${PN} = "-r restful"
 USERADD_PARAM_${PN} = "\
---system --home ${localstatedir}/lib/empty \
---no-create-home --shell /bin/false \
+--system --create-home --shell /bin/false \
```

Comment:
@mythi Updated the commit message.

Issue was that, iotivity-node instantiated by rest api server was failing to create configuration files under ${HOME}/.iotivity-node directory when the server started under non-root user (i.e restful). The reason for that was the restful user was created without a home directory. So, I changed --no-create-home option to --create-home to create user's home directory.